src/test/java/org/example/service/TicketServiceTest.java

```diff
@@ -0,0 +1,223 @@
+
+  @Mock private TicketRepository ticketRepository;
+  @InjectMocks private TicketService ticketService;
+
+  User user;
+  Ticket testTicket;
+
+  @BeforeEach
+  void init() {
+    user = new User("Masha", UserStatus.DEACTIVATED, new Timestamp(System.currentTimeMillis()));
+    testTicket = new Ticket(user, TicketType.DAY, new Timestamp(System.currentTimeMillis()));
+  }
+
+  @Test
+  void saveTicket_withTicket() {
+    testTicket.setId(UUID.randomUUID());
+
+    when(ticketRepository.save(testTicket)).thenReturn(testTicket);
+
+    Ticket result = ticketService.saveTicket(testTicket);
+
+    assertNotNull(result);
+    assertNotNull(testTicket.getId());
+    verify(ticketRepository, times(1)).save(testTicket);
+  }
+
+  @Test
+  void saveTicket_withNull() {
+    doThrow(new IllegalArgumentException()).when(ticketRepository).save(null);
+
+    assertThrows(IllegalArgumentException.class, () -> ticketService.saveTicket(null));
+    verify(ticketRepository, times(1)).save(null);
+  }
+
+  @Test
+  void saveTicket_withException_duringTransaction() {
+    doThrow(new OptimisticLockingFailureException("Save operation failed"))
+        .when(ticketRepository)
+        .save(testTicket);
+
+    assertThrows(
+        OptimisticLockingFailureException.class, () -> ticketService.saveTicket(testTicket));
+    verify(ticketRepository, times(1)).save(testTicket);
+  }
+
+  @Test
+  void getTicketById_withTicket() {
+    UUID id = UUID.randomUUID();
+
+    when(ticketRepository.findById(id)).thenReturn(Optional.ofNullable(testTicket));
+
+    Ticket actualResult = ticketService.getTicket(id);
+
+    assertEquals(testTicket, actualResult);
+    verify(ticketRepository, times(1)).findById(id);
+  }
+
+  @Test
+  void getTicketById_withNull() {
+    doThrow(new IllegalArgumentException()).when(ticketRepository).findById(null);
+
+    assertThrows(IllegalArgumentException.class, () -> ticketService.getTicket(null));
+    verify(ticketRepository, times(1)).findById(null);
+  }
+
+  @Test
+  void getTicketById_withException_duringTransaction() {
+    testTicket.setId(UUID.randomUUID());
+    doThrow(new OptimisticLockingFailureException("Id is invalid"))
+        .when(ticketRepository)
+        .findById(testTicket.getId());
+
+    assertThrows(
+        OptimisticLockingFailureException.class, () -> ticketService.getTicket(testTicket.getId()));
+    verify(ticketRepository, times(1)).findById(testTicket.getId());
+  }
+
+  @Test
+  void getTicketByUserID_withTicket() {
+    user.setId(UUID.randomUUID());
+    List<Ticket> expectedTickets = new ArrayList<>();
+    expectedTickets.add(
+        new Ticket(user, TicketType.DAY, new Timestamp(System.currentTimeMillis())));
+    expectedTickets.add(
+        new Ticket(user, TicketType.YEAR, new Timestamp(System.currentTimeMillis())));
+
+    when(ticketRepository.findByUserId(user.getId())).thenReturn(expectedTickets);
+
+    List<Ticket> result = ticketService.getTicketByUserId(user.getId());
```

Comment:
always give clear names for results: `actualTicketList` or `resultTicketlist` instead of just `result`

Reply:
Changed names of results actualTicketList and resultTicketlist